spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -971,4 +979,109 @@ public int hashCode() {
       return Objects.hashCode(values, uri, format);
     }
   }
+
+  @Nullable
+  public static ExecutorService executorService(int parallelism) {
+    return parallelism == 1 ? null : new ExecutorServiceFactory(parallelism);
+  }
+
+  private static class ExecutorServiceFactory implements ExecutorService, Serializable {
```

Comment:
Probably needs a rename since this doesn't actually make Executor Services, probably just LazyExcutorService?